sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusScalerProvider.cs

```diff
@@ -0,0 +1,116 @@
+            ILoggerFactory loggerFactory = serviceProvider.GetService<ILoggerFactory>();
+            ServiceBusMetadata serviceBusMetadata = JsonConvert.DeserializeObject<ServiceBusMetadata>(triggerMetadata.Metadata.ToString());
+            serviceBusMetadata.ResolveProperties(serviceProvider.GetService<INameResolver>());
+            ServiceBusClientFactory factory = new ServiceBusClientFactory(configuration, azureComponentFactory, messagingProvider, logForwarder, options);
+            ServiceBusAdministrationClient serviceBusAdminClient = factory.CreateAdministrationClient(serviceBusMetadata.Connection);
+            ServiceBusClient serviceBusClient = factory.CreateClientFromSetting(serviceBusMetadata.Connection);
+            ServiceBusReceiver serviceBusReceiver = !string.IsNullOrEmpty(serviceBusMetadata.QueueName) ? serviceBusClient.CreateReceiver(serviceBusMetadata.QueueName)
+            : serviceBusClient.CreateReceiver(serviceBusMetadata.TopicName, serviceBusMetadata.SubscriptionName);
+
+            _scaleMonitor = new ServiceBusScaleMonitor(
+                triggerMetadata.FunctionName,
+                serviceBusReceiver.EntityPath,
+                !string.IsNullOrEmpty(serviceBusMetadata.QueueName) ? ServiceBusEntityType.Queue : ServiceBusEntityType.Topic,
+                new Lazy<ServiceBusReceiver>(() => serviceBusReceiver),
+                new Lazy<ServiceBusAdministrationClient>(() => serviceBusAdminClient),
+                loggerFactory);
+
+            _targetScaler = new ServiceBusTargetScaler(
+                triggerMetadata.FunctionName,
+                serviceBusReceiver.EntityPath,
+                !string.IsNullOrEmpty(serviceBusMetadata.QueueName) ? ServiceBusEntityType.Queue : ServiceBusEntityType.Topic,
+                new Lazy<ServiceBusReceiver>(() => serviceBusReceiver),
+                new Lazy<ServiceBusAdministrationClient>(() => serviceBusAdminClient),
+                options.Value,
+                serviceBusMetadata.IsSessionsEnabled,
+                !string.Equals(serviceBusMetadata.Cardinality, "many", StringComparison.InvariantCultureIgnoreCase),
+                loggerFactory);
+        }
+
+        public IScaleMonitor GetMonitor()
+        {
+            return _scaleMonitor;
+        }
+
+        public ITargetScaler GetTargetScaler()
+        {
+            return _targetScaler;
+        }
+
+        internal class ServiceBusMetadata
+        {
+            [JsonProperty]
+            public string Type { get; set; }
+
+            [JsonProperty]
+            public string Connection { get; set; }
+
+            [JsonProperty]
+            public string QueueName { get; set; }
+
+            [JsonProperty]
+            public string TopicName { get; set; }
+
+            [JsonProperty]
+            public string SubscriptionName { get; set; }
+
+            [JsonProperty]
+            public bool IsSessionsEnabled { get; set; }
```

Comment:
We also need to add public bool [IsBatched](https://learn.microsoft.com/en-us/azure/azure-functions/functions-target-based-scaling?tabs=v5%2Ccsharp#:~:text=Execution%20Model-,IsBatched,-IsSessionsEnabled) for functions that are using batched processing

Reply:
There is already `Cardinality` property in the metadata, it looks like IsBatched is only for C# in-proc/isolated worker. I am confused why it was added in 5.x as we always can check if trigger input is singular or plural.

Reply:
I don't think this is exposed anywhere in 5.x for in-proc.